### DISCUSSION TITLE
different snippets for tex math environment for latex and orgmode documents

### DISCUSSION THREAD

**Discussion by fab6** [minad/tempel#123] [Q&A] [answered]
Hi,

I am using mostly yasnippets for equations in org and tex documents. There is an option to auto expand the template/snippet depending if one is active in an texmath environment with

```
# condition: (and (texmathp) 'auto)
```
or similar to auto expand if one is not in such an environment:
```
# condition: (and (not(texmathp)) 'auto)
```
Is it possible to auto expand in different environments with tempel as well? I did not find this in the documentation.

Thank you in advance!
Best Regards
Fab

**Comment by fab6** [comment]
Hi,
I assume that this is not possible yet with tempel or maybe my question was not clearly written. If so please let me know.
Thank you!
Best Regards

**Reply by minad** [reply]
I just didn't come to answer this question so far. I am currently going through the discussion boards.

**Comment by minad** [comment, marked as answer]
See https://github.com/minad/tempel#template-file-format for the `:when` key which can be used to specify a condition. Does this work for your use case?

**Comment by fab6** [comment]
ok, thank you for you help, I will check out, if I can figure it out

**Comment by fab6** [comment]
Hi,
small update, for me, this works good:
```
org-mode :when (org-inside-LaTeX-fragment-p)
(ff "\\frac{" p "}{" q "}")
(bb "\\mathbb{" p "}")
```
and
```
org-mode :when (not (org-inside-LaTeX-fragment-p))
(ato1  "#+attr_org: :width " (p "650") q)
```

Further I am using it with `aas` and `laas` to expand selected snippets automatically, e.g. like:

```
(aas-set-snippets 'org-mode
  "ato1"  (aas-tempel ato1)
  "atw1"  (aas-tempel atw1)
....
```
and
```
(aas-set-snippets 'laas-mode
  :cond #'texmathp 
  "fr"  (aas-tempel "\\frac{" p "}{" p "} " q)
....
```
with `aas-tempel` and help of some LLM
```
(defmacro aas-tempel (&rest elems)
  "Return a function that inserts a Tempel template for AAS/LAAS."
  (declare (indent defun))
  `(lambda ()
     (interactive)
     (require 'tempel)  ;; autoload safety
     ,(if (and (= (length elems) 1) (symbolp (car elems)))
          `(tempel-insert ',(car elems))     
        `(tempel-insert ',elems))))         
```
